tests/e2e-pw/tests/cart-block/cart-block.merchant.spec.ts

```diff
@@ -0,0 +1,190 @@
+			// blocks.
+			await page.evaluate(
+				"wc.blocksCheckout.registerCheckoutFilters( 'woo-test-namespace'," +
+					'{ additionalCartCheckoutInnerBlockTypes: ( value, extensions, { block } ) => {' +
+					"    value.push('core/table');" +
+					"    if ( block === 'woocommerce/cart-order-summary-block' ) {" +
+					"        value.push( 'core/audio' );" +
+					'    }' +
+					'    return value;' +
+					'}' +
+					'}' +
+					');'
+			);
+
+			await editor.insertBlock( { name: 'woocommerce/cart' } );
+			await expect(
+				page.locator( blockData.selectors.editor.block )
+			).toHaveCount( 1 );
+
+			// // Select the cart-order-summary-block block and try to insert a block. Check the Table block is available.
+			await editor.selectBlocks(
+				blockData.selectors.editor.block +
+					' [data-type="woocommerce/cart-order-summary-block"]'
+			);
+
+			const addBlockButton = await page
+				.getByRole( 'document', { name: 'Block: Order Summary' } )
+				.getByRole( 'button', { name: 'Add block' } );
+			await addBlockButton.click();
+			await page
+				.getByLabel( 'Search for blocks and patterns' )
+				.fill( 'Table' );
+			const tableButton = await page.getByRole( 'option', {
+				name: 'Table',
+			} );
+			await expect( tableButton ).toBeVisible();
+
+			await page
+				.getByLabel( 'Search for blocks and patterns' )
+				.fill( 'Audio' );
+
+			const audioButton = await page.getByRole( 'option', {
+				name: 'Audio',
+			} );
+			await test.expect( audioButton ).toBeVisible();
+
+			// Now check the filled cart block and expect only the Table block to be available there.
+			await editor.selectBlocks(
+				blockData.selectors.editor.block +
+					' [data-type="woocommerce/filled-cart-block"]'
+			);
+			const filledCartAddBlockButton = await page
+				.getByRole( 'document', { name: 'Block: Filled Cart' } )
+				.getByRole( 'button', { name: 'Add block' } )
+				.first();
+			await filledCartAddBlockButton.click();
+
+			const filledCartTableButton = await page.getByRole( 'option', {
+				name: 'Table',
+			} );
+			await expect( filledCartTableButton ).toBeVisible();
+
+			const filledCartAudioButton = await page.getByRole( 'option', {
+				name: 'Audio',
+			} );
+			await expect( filledCartAudioButton ).not.toBeVisible();
+		} );
+
+		test( 'shows empty cart when changing the view', async ( {
+			page,
+			editor,
+		} ) => {
+			await editor.selectBlocks( blockData.selectors.editor.block );
+			await page.getByRole( 'button', { name: 'Switch view' } ).click();
+			const emptyCartButton = await page.getByRole( 'menuitem', {
+				name: 'Empty Cart',
+			} );
+
+			// Focus the empty cart button and wait for the tooltip to disappear.
+			await emptyCartButton.focus();
+			await page.waitForTimeout( 1000 );
+			await emptyCartButton.click();
+
+			await expect(
+				page.locator(
+					blockData.selectors.editor.block +
+						' [data-type="woocommerce/filled-cart-block"]'
```

Comment:
Is there a reason why you didn't use `[getBlockByName](https://github.com/woocommerce/woocommerce-blocks/blob/fa3f9856629dcbf77d12fc9064b34b9820574776/tests/e2e-pw/utils/editor/EditorUtils.ts/#L19-L21)`?